webapp/channels/src/components/preparing_workspace/organization.tsx

```diff
@@ -0,0 +1,207 @@
+    updateTeam: (teamToUpdate: Team) => Promise<{error: string | null; updatedTeam: Team | null}>;
+    setInviteId: (inviteId: string) => void;
+    isSelfHosted: boolean;
+}
+
+const reportValidationError = debounce(() => {
+    trackEvent('first_admin_setup', 'validate_organization_error');
+}, 700, {leading: false});
+
+const Organization = (props: Props) => {
+    const {formatMessage} = useIntl();
+    const dispatch = useDispatch();
+
+    const [triedNext, setTriedNext] = useState(false);
+    const inputRef = useRef<HTMLInputElement>();
+    const validation = teamNameToUrl(props.organization || '');
+    const teamApiError = useRef<typeof TeamApiError | null>(null);
+
+    useEffect(props.onPageView, []);
+
+    const teams = useSelector(getActiveTeamsList);
+    useEffect(() => {
+        if (!teams) {
+            dispatch(getTeams(0, 60));
+        }
+    }, [teams]);
+
+    const setApiCallError = () => {
+        teamApiError.current = TeamApiError;
+    };
+
+    const updateTeamNameFromOrgName = async () => {
+        if (!inputRef.current?.value) {
+            return;
+        }
+        const name = inputRef.current?.value.trim();
+
+        const currentTeam = teams[0];
+
+        if (currentTeam && name && name !== currentTeam.display_name) {
+            const {error} = await props.updateTeam({...currentTeam, display_name: name});
+            if (error !== null) {
+                setApiCallError();
+            }
+        }
+    };
+
+    const createTeamFromOrgName = async () => {
+        if (!inputRef.current?.value) {
+            return;
+        }
+        const name = inputRef.current?.value.trim();
+
+        if (name) {
+            const {error, newTeam} = await props.createTeam(name);
+            if (error !== null || newTeam === null) {
+                props.setInviteId('');
+                setApiCallError();
+                return;
+            }
+            props.setInviteId(newTeam.invite_id);
+        }
+    };
+
+    const handleOnChange = (e: ChangeEvent<HTMLInputElement>) => {
+        props.setOrganization(e.target.value);
+        teamApiError.current = null;
+    };
+
+    const onNext = (e?: React.KeyboardEvent | React.MouseEvent) => {
+        if (e && (e as React.KeyboardEvent).key) {
+            if ((e as React.KeyboardEvent).key !== Constants.KeyCodes.ENTER[0]) {
+                return;
+            }
+        }
+        if (!triedNext) {
+            setTriedNext(true);
+        }
+
+        // if there is already a team, maybe because a page reload, then just update the teamname
+        const thereIsAlreadyATeam = teams.length > 0;
+        teamApiError.current = null;
+
+        if (!validation.error && !thereIsAlreadyATeam && props.isSelfHosted) {
+            createTeamFromOrgName();
+        } else if (!validation.error && thereIsAlreadyATeam && props.isSelfHosted) {
+            updateTeamNameFromOrgName();
+        }
```

Comment:
These checks for `isSelfHosted` doesn't hurt but I believe this screen is not even shown if not selfhosted

Reply:
you are right, these can be removed.